ci/go-ossf-slsa3-publish.yml

```diff
@@ -0,0 +1,35 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This workflow lets you compile your Go project using a SLSA3 compliant builder. 
+# This workflow will generate a so-called "provenance" file describing the steps 
+# that were performed to generate the final binary.
+# The project is an initiative of the OpenSSF (openssf.org) and is developed at 
+# https://github.com/slsa-framework/slsa-github-generator.
+# The provenance file can be verified using https://github.com/slsa-framework/slsa-verifier.
+# For more information about SLSA and how it improves the supply-chain, visit slsa.dev.
+
+name: SLSA Go releaser
+on:
+  workflow_dispatch:
+  release:
+    types: [created]
+
+permissions: read-all
+
+jobs:
+  build:
+    permissions:
+      id-token: write # To sign.
+      contents: write # To upload release assets.
+      actions: read   # To read workflow path.
+    # If you need more configuration options, such as ldflag examples, 
+    # visit https://github.com/slsa-framework/slsa-github-generator#golang-projects.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.0.0
```

Comment:
Not a comment for this PR but something to think about for future in case you plan to add more workflows for building other languages. Can they be encapsulated in a single workflow rather than a separate one for each language?
For example, look at [codeql](https://github.com/actions/starter-workflows/blob/874ee449ed7c7203faec42e7708b2929d123ff2b/code-scanning/properties/codeql.properties.json#L6) one which caters to multiple languages.

Reply:
Thanks for the tip. I was initially thinking of having different workflows for each language. Since most projects contain a single "main" project, it would avoid noise. I'll think about this for future languages, thanks!